src/mame/includes/tsconf.h

```diff
@@ -0,0 +1,212 @@
+// license:BSD-3-Clause
+// copyright-holders:Andrei I. Holub
+/*****************************************************************************
+ *
+ * includes/tsconfig.h
+ *
+ ****************************************************************************/
+#ifndef MAME_INCLUDES_TSCONF_H
+#define MAME_INCLUDES_TSCONF_H
+
+#pragma once
+
+#include "emu.h"
+#include "machine/beta.h"
+#include "machine/spi_sdcard.h"
+#include "spectrum.h"
+#include "spec128.h"
+#include "machine/tsconfdma.h"
+#include "machine/nvram.h"
+#include "machine/pckeybrd.h"
+#include "machine/8042kbdc.h"
+#include "tilemap.h"
+
+#define LOG_ERROR (1U << 0)
+#define LOG_WARN (1U << 1)
+#define LOG_DEBUG (1U << 2)
+
+#define VERBOSE (LOG_WARN)
+#define LOG_OUTPUT_FUNC osd_printf_info
+
+#include "logmacro.h"
+#define LOGERROR(...) LOGMASKED(LOG_DEBUG | LOG_WARN | LOG_ERROR, __VA_ARGS__)
+#define LOGWARN(...) LOGMASKED(LOG_DEBUG | LOG_WARN, __VA_ARGS__)
+#define LOGDEBUG(...) LOGMASKED(LOG_DEBUG, __VA_ARGS__)
```

Comment:
You’re not supposed to put logmacro.h glue in headers – it makes life hard if anyone wants to derive anything from any of your classes.

Additionally, this goes against how logmacro.h is supposed to be used – you’re supposed to define bits for *categories*, not *levels*.  It’s far more useful to be able to enable or disable messages by category or aspect rather than grouping in arbitrary levels.  Bit 0 is reserved for the default `LOG` category `LOG_GENERAL` and for things that are actually errors, you should just call `logerror` unconditionally.

Reply:
I think I'll get rid of all log macros in this PR. Implemented way was just more comfortable for me.
logerror must be enough for desired cases.